control-plane/stor-port/src/types/v0/transport/nexus.rs

```diff
@@ -79,6 +79,93 @@ impl From<Nexus> for models::Nexus {
 
 rpc_impl_string_uuid!(NexusId, "UUID of a nexus");
 
+/// A gRPC request type for creating snapshot of a nexus, which essentially
+/// means a snapshot of all(or selected) replicas associated with that nexus.
+pub struct CreateNexusSnapshot {
+    /// UUID of the nexus.
+    nexus: NexusId,
+    /// UUID of the entity, nexus in this case.
+    entity: uuid::Uuid,
```

Comment:
We already have the nexus id above, so this is not needed?

Reply:
I guess I should modify the comment to not say nexus. entity can be potentially set as volume id ?

Reply:
Yes.. could be anything for now... let's make these String for now.